### PR TITLE
fix(DateRange): update format and condition in renderComponent

### PR DESCRIPTION
**Pull request by PrincAm**

### CODE REVIEW

File: src/components/Filters/FilterRow/FilterRow.tsx [hunk 1]
```diff
@@ -89,7 +88,8 @@ const getOperatorOptions = curry((operatorValue) =>
 
 const getFieldOptions = map(normalizeOptions);
 
-const isArrayOfOptionObjects = when(isArray, pipe(head, has('value')));
+const isArrayOfOptionObjects = (value) =>
+  isArray(value) && pipe(head, has('value'))(value);
```
Comment (PrincAm) [inline src/components/Filters/FilterRow/FilterRow.tsx:92]: @ajkl2533 Could be improved? I need to return false if argument is not array

Comment (ajkl2533) [reply src/components/Filters/FilterRow/FilterRow.tsx:92]: you can do
```
const isArrayOfOptionObjects = R.both(
 RA.isArray,
 R.pipe(R.head, R.has('value'))
)
```
According to Ramda docs
```
R.both
...
Note that this is short-circuited, meaning that the second function will not be invoked if the first returns a false-y value. ...

File: src/components/Filters/components/DateRangePicker/DateRangePicker.tsx [hunk 2]
```diff
@@ -35,6 +35,7 @@ const DateRangePicker: React.FC<DateRangePickerProps> = ({
       <DatePicker
         calendarClassName="DateRangePicker-calendar"
         className="DateRangePicker-input"
+        dateFormat="d MMM, yyyy"
```
Comment (ajkl2533) [inline src/components/Filters/components/DateRangePicker/DateRangePicker.tsx:38]: So what are the valid input formats? Can I use different formats or just this one?

File: src/components/Filters/FilterRow/FilterRow.tsx (hunk 1 above)
Comment (PrincAm) [reply src/components/Filters/FilterRow/FilterRow.tsx]: `R.both` this is it! Perfect

File: src/components/Filters/components/DateRangePicker/DateRangePicker.tsx (hunk 2 above)
Comment (PrincAm) [reply src/components/Filters/components/DateRangePicker/DateRangePicker.tsx:38]: `d MMM, yyyy` is display format and you can input any of these: `MM/DD/YYYY, YYYY/MM/DD, feb 2021, 12 feb 2021, 2021 feb 12, 2021 12 feb`. Parsing itself is done by library so it is much less work than I thought. Even more, we found out Atlas is using same library and ours is most likely first date picker in platform. The refactoring paid off in the end.

**Review by ajkl2533 — APPROVED** [review]

**Pull request merged**